File: trunk/regexwikinoconvert.cpp
#include <QDebug>
#include <QStringList>
#include "regexwikinoconvert.h"
#include "config.h"


RegexWikiNoConvert::RegexWikiNoConvert(QString &source)
    : Regex(source)
{
    QString pattern = "(?:\\`\\{\\`\\{\\s*" +
            QString(WIKI_NO_CONVERT_TAG) +
            "\\s*\\|\\s*((?:[^\\`]|\\`\\`)*)"
            "\\`\\}\\`\\})";
    regexp.setPattern(pattern);
}


QString RegexWikiNoConvert::getMatchEquivalent()
{
    qDebug() << "NOCONVERT template Found:" << regexp.capturedTexts();
    //Return the captured text, which is the first parameter of template:
    return regexp.cap(1);
}

File: phpport/lib/regexwikinoconvert.php
<?php

/*!
 * @file regexwikinoconvert.h
 * @brief This file contains declaration of RegexWikiNoConvert class.
 * @author Mousa Moradi
 */

include_once('regex.php');
include_once('config.php');


/*!
 * @brief The Wiki NoConvert template checker Regex class.
 */
class RegexWikiNoConvert extends Regex
{
    /*!
     * @brief Class constructor
     * @param source The source text to work on it
     * @param convertor The convertor engine which may be used to convert some pieces
     */
    public function __construct()
    {
		global $WIKI_NO_CONVERT_TAG;
		$this->pattern = "(?:\\`\\{\\`\\{\\s*" .
						 $WIKI_NO_CONVERT_TAG .
						 "\\s*\\|\\s*((?:[^\\`]|\\`\\`)*)" .
						 "\\`\\}\\`\\})";
        $this->pattern = '#' . $this->pattern . '#';
	}


    /*!
     * @brief Reads the regexp.cap() and gets the equivalent for the matched substrings
     * @return The equivalent for the matched substrings. It may be the same as first matched one.
     */
    protected function getMatchEquivalent()
    {
		$count = count($this->matches[0]);
		for($i=0; $i<$count; $i++)
		{
			//Set the first parameter as result:
			$this->matches['equivalent'][$i] = $this->matches[1][$i];
		}
	}
}

?>
